lib/annotations2triannon/open_annotation.rb

```diff
@@ -261,6 +254,33 @@ def to_ttl
       @graph.dump(:ttl, standard_prefixes: true)
     end
 
+    private
+
+    # Sets the annotation ID as an RDF::URI from id, but if the id is nil, it
+    # will try to get the ID from the graph and, as a last resort, it will
+    # generate a unique ID using a UUID.
+    # @param id [UUID|URI|String] to identify an open annotation
+    def set_id(id)
+      raise ArgumentError, 'id cannot be an empty String' if (id.instance_of?(String) && id.empty?)
+      if id.nil?
+        # Try to get the ID from the graph or use a UUID
+        q = [nil, RDF.type, OA.Annotation]
+        @id = @graph.query(q).collect {|s| s.subject }.first || RDF::URI.parse(UUID.generate)
```

Comment:
Surely not for the top-most resource in a POSTed RDFSource? How would you ever get it back? That would be even more moronic than their moronic-but-vaguely-understandable decision to keep internal blank nodes.  So as Triannon builds containers underneath the annotation for the body and target, everything should be assigned a URI. It might be good to do a walk through of the client<-->triannon, and triannon<-->F4 interactions to verify they are as expected.
c.f. "Open Annotation" interpretation of Drew :S